tests/blake2-arm-test.c

```diff
@@ -8,439 +8,100 @@
-  Hacl_Blake2b_32_blake2b(64,comp_b,blake2b_test1_size_plaintext,blake2b_test1_plaintext,blake2b_test1_size_key,blake2b_test1_key);
-  printf("computed:");
-  for (int i = 0; i < 64; i++)
-    printf("%02x",comp_b[i]);
-  printf("\n");
-  printf("expected:");
-  for (int i = 0; i < 64; i++)
-    printf("%02x",blake2b_test1_expected[i]);
-  printf("\n");
-  ok = true;
-  for (int i = 0; i < 64; i++)
-    ok = ok & (blake2b_test1_expected[i] == comp_b[i]);
-  if (ok) printf("Success!\n");
-  else printf("FAILED!\n");
-  all_ok = all_ok && ok;
-
-  Hacl_Blake2b_32_blake2b(64,comp_s,blake2b_test13_size_plaintext,blake2b_test13_plaintext,blake2b_test13_size_key,blake2b_test13_key);
-  printf("computed:");
-  for (int i = 0; i < 64; i++)
-    printf("%02x",comp_s[i]);
-  printf("\n");
-  printf("expected:");
-  for (int i = 0; i < 64; i++)
-    printf("%02x",blake2b_test13_expected[i]);
-  printf("\n");
-  ok = true;
-  for (int i = 0; i < 64; i++)
-    ok = ok & (blake2b_test13_expected[i] == comp_s[i]);
-  if (ok) printf("Success!\n");
-  else printf("FAILED!\n");
-  all_ok = all_ok && ok;
+int main()
+{
+  int in_len = vectors2b[0].input_len;
+  uint8_t *in = vectors2b[0].input;
+  int key_len = vectors2b[0].key_len;
+  uint8_t *key = vectors2b[0].key;
+  int exp_len = vectors2b[0].expected_len;
+  uint8_t *exp = vectors2b[0].expected;
+
+  int in_len1 = vectors2b[1].input_len;
+  uint8_t *in1 = vectors2b[1].input;
+  int key_len1 = vectors2b[1].key_len;
+  uint8_t *key1 = vectors2b[1].key;
+  int exp_len1 = vectors2b[1].expected_len;
+  uint8_t *exp1 = vectors2b[1].expected;
+
+  bool ok = print_test2b(in_len,in,key_len,key,exp_len,exp);
+  ok = print_test2b(in_len1,in1,key_len1,key1,exp_len1,exp1) && ok;
+
+  int in_len2 = vectors2s[0].input_len;
+  uint8_t *in2 = vectors2s[0].input;
+  int key_len2 = vectors2s[0].key_len;
+  uint8_t *key2 = vectors2s[0].key;
+  int exp_len2 = vectors2s[0].expected_len;
+  uint8_t *exp2 = vectors2s[0].expected;
+
+  int in_len3 = vectors2s[1].input_len;
+  uint8_t *in3 = vectors2s[1].input;
+  int key_len3 = vectors2s[1].key_len;
+  uint8_t *key3 = vectors2s[1].key;
+  int exp_len3 = vectors2s[1].expected_len;
+  uint8_t *exp3 = vectors2s[1].expected;
+
+  int in_len4 = vectors2s[2].input_len;
+  uint8_t *in4 = vectors2s[2].input;
+  int key_len4 = vectors2s[2].key_len;
+  uint8_t *key4 = vectors2s[2].key;
+  int exp_len4 = vectors2s[2].expected_len;
+  uint8_t *exp4 = vectors2s[2].expected;
+
+  int in_len5 = vectors2s[3].input_len;
+  uint8_t *in5 = vectors2s[3].input;
+  int key_len5 = vectors2s[3].key_len;
+  uint8_t *key5 = vectors2s[3].key;
+  int exp_len5 = vectors2s[3].expected_len;
+  uint8_t *exp5 = vectors2s[3].expected;
+
+  ok = print_test2s(in_len2,in2,key_len2,key2,exp_len2,exp2) && ok;
+  ok = print_test2s(in_len3,in3,key_len3,key3,exp_len3,exp3) && ok;
+  ok = print_test2s(in_len4,in4,key_len4,key4,exp_len4,exp4) && ok;
+  ok = print_test2s(in_len5,in5,key_len5,key5,exp_len5,exp5) && ok;
```

Comment:
minor remark: you could do
```
bool ok = true;
for (int i = 0; i < vectors2b/sizeof(blake2b_test_vector); ++i) {
  ok &= print_test2b(vectors2b[i].input, ...);
}
// same for vectors2s
```

that way if someone adds more test vectors, no need to change the test for them to be taken into account